github/client.go

```diff
@@ -24,6 +24,15 @@ func NewClient() *Client {
 	return &Client{}
 }
 
+func (c *Client) GetGithubRepoFromRawURL(rawURL string) (svcmodel.GithubRepo, error) {
```

Comment:
After the GitHub refactoring, this method will call a GitHub API to check if the repository exists and is accessible with the provided token.